File: leetcode/0014-longest-common-prefix/solution.cpp
class Solution {
    public:
        string longestCommonPrefix(vector<string>& strs) {
            if (strs.empty()) {
                return "";
            } 
            
            sort(strs.begin(), strs.end());
            auto first = strs.front();
            auto last = strs.back();
    
            int i = 0;
            while (i < first.length() && i < last.length() && first[i] == last[i]){
                i++;
            }
            
            return first.substr(0, i);
        }
};

File: leetcode/0014-longest-common-prefix/solution.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        if not strs:
            return False
        
        strs = sorted(strs)
        first, last = strs[0], strs[-1]

        i = 0
        while i < len(first) and i < len(last) and first[i] == last[i]:
            i += 1
            
        return first[:i]